### PR TITLE
[CI] Added workflow for publishing rustdoc.

### PR DESCRIPTION
**Pull request by apepkuss**
In this PR, a new CI workflow is implemented for publishing the rustdoc of  the `wasmedge-sys` crate to github pages.

To enable the Github Pages feature, it is required to specify the `Source` as shown in the red rectangle below:

image

### References
  - [actions-gh-pages](https://github.com/peaceiris/actions-gh-pages#%EF%B8%8F-deploy-to-external-repository-external_repository)
  - [rust-analyzer rustdoc.yaml](https://github.com/rust-analyzer/rust-analyzer/blob/master/.github/workflows/rustdoc.yaml)
  - [Prepare your Rust API docs for Github Pages](https://dev.to/deciduously/prepare-your-rust-api-docs-for-github-pages-2n5i)

### CODE REVIEW

File: .github/workflows/wasmedge-sys-docs.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: wasmedge-sys-docs
+
+concurrency:
+  group: rust-crate-release-${{ github.head_ref }}
+  cancel-in-progress: true
+
+on:
```
Comment (hydai) [inline .github/workflows/wasmedge-sys-docs.yml:7]: Maybe we need to test the docs generation on master when the related rust files change. But we don't publish it until emitting the release tag.

**Review by hydai — CHANGES_REQUESTED** [review]

File: .github/workflows/wasmedge-sys-docs.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+name: wasmedge-sys-docs
+
+concurrency:
+  group: rust-crate-release-${{ github.head_ref }}
```
[the remark this answers is not in the document]
Comment (apepkuss) [reply .github/workflows/wasmedge-sys-docs.yml:4]: Accepted. I'll fix the issue. Thanks!

File: .github/workflows/wasmedge-sys-docs.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  cancel-in-progress: true
+
+on:
+  push:
+    tags:
+      - "wasmedge-sys/[0-9]+.[0-9]+.[0-9]+*"
```
[the remark this answers is not in the document]
Comment (apepkuss) [reply .github/workflows/wasmedge-sys-docs.yml:10]: The original tag pattern is only for `wasmedge-sys`. Once `wasmedge-rs` is introduced, the original tag could be ambiguous. Based on this consideration, suggest using new tag names for the existed `wasmedge-sys` and the upcoming `wasmedge-rs`, respectively. `wasmedge-sys/[0-9]+.[0-9]+.[0-9]+*` is used for `wasmedge-sys`.

File: .github/workflows/wasmedge-sys-docs.yml (hunk 1 above)
Comment (apepkuss) [reply .github/workflows/wasmedge-sys-docs.yml:7]: You're right. Using this strategy is based on the following consideration.  The current workflow is only triggered when there is a new tag coming, like `wasmedge-sys/0.4.0`. Such a setting can guarantee that the crate published on crates.io and the rustdoc published on the Github pages are built from the same commit. However, if we used `push-on-master` to trigger the workflow, the rustdoc would be built from the latest rust code, but not from the published version, that is, there would be version inconsistency between the published crate and the published rustdoc.

File: .github/workflows/wasmedge-sys-docs.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
+name: wasmedge-sys-docs
+
+concurrency:
+  group: wasmedge-sys-docs-${{ github.head_ref }}
+  cancel-in-progress: true
+
+on:
+  push:
+    tags:
+      - "wasmedge-sys/[0-9]+.[0-9]+.[0-9]+*"
```
Comment (hydai) [inline .github/workflows/wasmedge-sys-docs.yml:10]: Hmmm. The previous tag style will be `rust/version`. Do you want to change it to `wasmedge-sys/version`? If so, then you will need to update the rust related workflows.
截圖 2022-03-10 at 16 03 39@2x

**Review by hydai — CHANGES_REQUESTED** [review]

**Review by hydai — APPROVED** [review]

**Comment by apepkuss** [comment]
@hydai Thanks for the review!

**Pull request merged**